### PR TITLE
feat: add image package deployment

### PR DESCRIPTION
**Pull request by kapas2004**
- add package_type and image_uri variables
- modify lambda function to accept image package deployment

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -159,16 +159,18 @@ resource "aws_lambda_function" "default" {
   architectures                  = [var.architecture]
   code_signing_config_arn        = var.code_signing_config_arn
   description                    = var.description
-  filename                       = var.s3_bucket == null ? local.filename : null
+  filename                       = var.s3_bucket == null ? var.image_uri == null ? local.filename : null : null
```
Comment (shoekstra) [inline main.tf:162]:
```suggestion
  filename                       = var.s3_bucket == null || var.image_uri == null ? local.filename : null
```

We can skip the nested operator here.

**Review by shoekstra — DISMISSED** [review]

File: main.tf (hunk 1 above)
Comment (kapas2004) [reply main.tf]: Change applied

File: variables.tf [hunk 2]
```diff
@@ -110,6 +116,12 @@ variable "name" {
   description = "The name of the lambda"
 }
 
+variable "package_type" {
+  type        = string
+  default     = null
+  description = "The Lambda deployment package type. Valid options: Zip or Image"
```
[the remark this answers is not in the document]
Comment (kapas2004) [reply variables.tf]: Zip and Image changed to lower case

**Comment by kapas2004** [comment]
Is there any blocker this PR to be merged?

**Comment by marwinbaumannsbp** [comment]
stale

**Pull request closed without merging**